nautobot_golden_config/jobs.py

```diff
@@ -22,11 +22,28 @@
 name = "Golden Configuration"  # pylint: disable=invalid-name
 
 
-def get_refreshed_repos(job_obj, repo_type):
+def get_refreshed_repos(job_obj, repo_type, devices=None, gc_settings=None):
     """Small wrapper to pull latest branch, and return a GitRepo plugin specific object."""
-    repository_records = set(
-        getattr(gcs, repo_type) for gcs in GoldenConfigSetting.objects.all() if getattr(gcs, repo_type, None)
-    )
+    if isinstance(devices, Device):
+        # AllGoldenConfig only accepts a single device object, so we cast it to a list here for consistency
+        devices = [devices]
+    gc_settings_query = gc_settings if gc_settings else GoldenConfigSetting.objects.all()
+
+    if devices and all(isinstance(d, Device) for d in devices):
+        # Only return the repositories that have the device(s) in scope (#271)
+        device_ids = [device.id for device in devices]
+        repository_records = set(
```

Comment:
Curious if there might be an equality check we can do here.  We know all the GCS, if we for instance had 5 settings and we selected 100 devices to run this on, we could evaluate after each iteration that we don't already have all 5 settings in the list, if we did there would be no reason to continue with this loop.